doc/build/snippets/design.rst

```diff
@@ -0,0 +1,61 @@
+Snippets Design
+###############
+
+This page documents design considerations related to snippets.
+
+List of Design Goals
+********************
+
+As design goals, snippets are:
+
+- **extensible**: for example, it is possible to add board support for an
+  existing built-in snippet without modifying the zephyr repository
+
+- **composable**: it is possible to use multiple snippets at once, for example
+  using:
+
+  .. code-block:: console
+
+     west build -s <snippet1> -s <snippet2> ...
+
+- **able to combine multiple types of configuration**: for example, it is
+  possible to specify Kconfig fragments (.conf files) and devicetree overlays
+  (.overlay files) which should be applied at the same time, along with doing
+  other things, like providing additional Kconfig.defconfig settings
+
+- **specializable**: for example, it is possible to customize a snippet's
+  behavior for a particular board, or board revision
+
+- **future-proof and backwards-compatible**: arbitrary future changes to the
+  snippets feature will be possible without breaking backwards compatibility
+  for older snippets
+
+- **applicable to purely "software" changes**: unlike the shields feature,
+  snippets do not assume the presence of a "daughterboard", "shield", "hat", or
+  any other type of external assembly which is connected to the main board
+
+- **DRY** (don't repeat yourself): snippets allow you to skip unnecessary
+  repetition, like copy/pasting a snippet's configuration settings for a
+  vendor development board into a separate board-specific configuration file
+  for a product
+
+Comparison with Shields
+***********************
+
+Snippets were originally developed to bring the :ref:`shields` feature set to a
+wider context, and to extend some of the shields feature's design limitations.
+As a result, snippets are a strict generalization of shields: any shield can be
+specified as a snippet, but the reverse is not true: not all snippets can be
+"expressed" as a shield.
+
+Here is a list of some limitations of the shields feature not present in the
+snippets feature:
+
+- **limited to external ICs**: at least in concept, shields require
+  some sort of external hardware
+- **not extensible**: adding board support for a shield defined in
+  :zephyr_file:`zephyr/boards/shields/` requires modifying the zephyr
+  repository itself
+- **not DRY**: for example, shields require identical board-specific
+  devicetree overlays to be specified via multiple identical files on
+  the file system
```

Comment:
This is not right, shields does not require identical board-specific devicetree overlays, but one can customize it for a specific board, see [Board compatibility](https://docs.zephyrproject.org/latest/hardware/porting/shields.html#board-compatibility) and  [Board specific shield configuration](https://docs.zephyrproject.org/latest/hardware/porting/shields.html#board-specific-shield-configuration).